src/tensor/mod.rs

```diff
@@ -111,15 +111,13 @@
 //! zip archives.
 
 pub(crate) mod cpu;
-mod tensor_impls;
-
 #[cfg(feature = "cuda")]
 pub(crate) mod cuda;
-
 #[cfg(feature = "numpy")]
 pub(crate) mod numpy;
-
 pub(crate) mod storage_traits;
+mod tensor_impls;
+pub(crate) mod visitors; // TODO pub?
```

Comment:
note to self

Reply:
To me, the visitors module feels like it belongs in `nn`, because its main utility is in operating on nn modules.

Reply:
That's what I did initially, however optim now also uses the visitors. optim depending on nn could makes sense the more I think about it...

Reply:
Moved